### ISSUE TITLE
Ty allows instantiation of abstract classes with unimplemented methods (false negative)

### ISSUE THREAD

**Issue by benedekaibas** [astral-sh/ty#1405]
### Summary

Pyre fails to detect when a `Final` class attribute is overridden by a property in a subclass while other type checkers detect it correctly. I have tested this code on various type checkers including: `zuban`, `pyrefly`, `ty`, and `mypy`

```python
from typing import Final

class Parent:
    id: Final[str] = "xyz"

class Child(Parent):
    @property
    def id(self) -> str:  # Overriding Final
        return "abc"

if __name__ == "__main__":
    print(Child().id)
```

### Version

ty 0.0.1-alpha.23

This issue might be worth taking a look at in my point of view!

### Version

ty 0.0.1-alpha.23

**Comment by AlexWaygood** [comment]
Hmm, the body of your issue seems to be about something different to what the title of your issue would imply

**Comment by benedekaibas** [comment]
> Hmm, the body of your issue seems to be about something different to what the title of your issue would imply

Sorry for that, I have pasted the wrong code example since I have more. The correct title should be: Ty allows Final class attribute to be overridden by property

I have changed it to the correct title. Sorry for the confusion!

**Comment by AlexWaygood** [comment]
thanks!

**Comment by AlexWaygood** [comment]
Looking at this again, I think this is the same as https://github.com/astral-sh/ty/issues/871
